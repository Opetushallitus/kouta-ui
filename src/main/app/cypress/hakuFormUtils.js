import merge from 'lodash/merge';

import organisaatio from './data/organisaatio';
import {
  stubKoodistoRoute,
  stubHakemuspalveluLomakkeetRoute,
  stubOppijanumerorekisteriHenkiloRoute,
} from './utils';

export const stubHakuFormRoutes = ({ cy, organisaatioOid }) => {
  cy.server();

  cy.route({
    method: 'GET',
    url: `**/organisaatio-service/rest/organisaatio/v4/${organisaatioOid}**`,
    response: merge(organisaatio(), {
      oid: organisaatioOid,
    }),
  });

  cy.route({
    method: 'POST',
    url: '**/organisaatio-service/rest/organisaatio/v4/findbyoids',
    response: [
      merge(organisaatio(), {
        oid: organisaatioOid,
      }),
    ],
  });

  stubKoodistoRoute({ koodisto: 'haunkohdejoukko', cy });
  stubKoodistoRoute({ koodisto: 'hakutapa', cy });
  stubKoodistoRoute({ koodisto: 'kausi', cy });
  stubKoodistoRoute({ koodisto: 'opetuspaikkakk', cy });
  stubKoodistoRoute({ koodisto: 'kausi', cy });

  cy.route({
    method: 'GET',
    url: '**/haku/list**',
    response: [],
  });

  stubHakemuspalveluLomakkeetRoute({ cy });
  stubOppijanumerorekisteriHenkiloRoute({ cy });
};
