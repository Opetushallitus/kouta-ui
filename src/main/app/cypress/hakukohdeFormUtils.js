import merge from 'lodash/merge';

import { stubKoodistoRoute } from './utils';
import organisaatio from './data/organisaatio';
import haku from './data/haku';

export const stubHakukohdeFormRoutes = ({ cy, organisaatioOid, hakuOid }) => {
  cy.server();

  cy.route({
    method: 'GET',
    url: `**/organisaatio-service/rest/organisaatio/v4/${organisaatioOid}**`,
    response: merge(organisaatio(), {
      oid: organisaatioOid,
    }),
  });

  stubKoodistoRoute({ koodisto: 'pohjakoulutusvaatimustoinenaste', cy });
  stubKoodistoRoute({ koodisto: 'kausi', cy });
  stubKoodistoRoute({ koodisto: 'valintakokeentyyppi', cy });
  stubKoodistoRoute({ koodisto: 'liitetyypitamm', cy });

  cy.route({
    method: 'GET',
    url: `**/haku/${hakuOid}`,
    response: merge(haku(), {
      oid: hakuOid,
      organisaatioOid: organisaatioOid,
    }),
  });

  cy.route({
    method: 'GET',
    url: `**/hakukohde/list**`,
    response: [],
  });
};
