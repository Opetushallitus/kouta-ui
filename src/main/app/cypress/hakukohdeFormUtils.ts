import { merge } from 'lodash';

import {
  stubKoodistoRoute,
  stubHakemuspalveluLomakkeetRoute,
  stubOppijanumerorekisteriHenkiloRoute,
  stubKoodiRoute,
  stubCommonRoutes,
  jatka,
} from '#/cypress/utils';

import haku from '#/cypress/data/haku';
import hakukohde from '#/cypress/data/hakukohde';
import createKoodi from '#/cypress/data/koodi';
import koulutus from '#/cypress/data/koulutus';
import toteutus from '#/cypress/data/toteutus';
import valintaperuste from '#/cypress/data/valintaperuste';
import { addMockFileRoutes } from './mockUtils';

const toimipisteTarjoajat = [
  '1.2.246.562.10.16538823663',
  '1.2.246.562.10.2013081415065445951365',
  '1.2.246.562.10.20485193278',
  '1.2.246.562.10.20866993056',
  '1.2.246.562.10.23017513880',
  '1.2.246.562.10.45854578546',
  '1.2.246.562.10.55355715099',
  '1.2.246.562.10.55711304158',
  '1.2.246.562.10.56139411567',
  '1.2.246.562.10.60465978314',
  '1.2.246.562.10.656909794910',
  '1.2.246.562.10.78321186062',
  '1.2.246.562.10.879177258610',
  '1.2.246.562.10.93115250668',
];

const selectedToimipisteNimi = /stadin ammatti- ja aikuisopisto, myllypuron toimipaikka/i;

const selectedToimipiste = '1.2.246.562.10.45854578546';

export const fillJarjestyspaikkaSection = ({ jatka: jatkaProp = true }) => {
  cy.findByTestId('jarjestyspaikkaOidSection').within(() => {
    toimipisteTarjoajat.forEach(oid => {
      cy.get(`[value="${oid}"]`).should('exist');
      if (oid === selectedToimipiste) {
        cy.get(`[value="${oid}"]`).should('not.be.disabled');
      } else {
        cy.get(`[value="${oid}"]`).should('be.disabled');
      }
    });

    cy.findByText(selectedToimipisteNimi).click();
    if (jatkaProp) {
      jatka();
    }
  });
};

export const prepareTest = ({
  tyyppi,
  hakuOid,
  hakukohdeOid,
  organisaatioOid,
  edit = false,
  tarjoajat,
}) => {
  const toteutusOid = '2.1.1.1.1.1';
  const koulutusOid = '3.1.1.1.1';
  const valintaperusteId = '649adb37-cd4d-4846-91a9-84b58b90f928';

  const testHakukohdeFields = {
    toteutusOid,
    hakuOid,
    organisaatioOid,
    oid: hakukohdeOid,
    valintaperusteId,
  };

  cy.server();

  addMockFileRoutes('hakukohde.mock.json');
  stubHakukohdeFormRoutes({ organisaatioOid, hakuOid });

  cy.route({
    method: 'GET',
    url: `**/toteutus/${toteutusOid}`,
    response: merge(toteutus({ tyyppi }), {
      oid: toteutusOid,
      organisaatioOid,
      koulutusOid: koulutusOid,
      tila: 'julkaistu',
      tarjoajat,
    }),
  });

  cy.route({
    method: 'GET',
    url: `**/koulutus/${koulutusOid}`,
    response: merge(koulutus({ tyyppi }), {
      oid: koulutusOid,
      tarjoajat,
      organisaatioOid: organisaatioOid,
      tila: 'julkaistu',
    }),
  });

  cy.route({
    method: 'GET',
    url: '**/valintaperuste/list**',
    response: [
      merge(valintaperuste(), {
        id: valintaperusteId,
        nimi: { fi: 'Valintaperusteen nimi' },
        tila: 'julkaistu',
      }),
    ],
  });

  cy.route({
    method: 'GET',
    url: `**/valintaperuste/${valintaperusteId}`,
    response: merge(valintaperuste(), {
      id: valintaperusteId,
      nimi: { fi: 'Valintaperusteen nimi' },
      tila: 'julkaistu',
    }),
  });

  cy.route({
    method: 'GET',
    url: `**/hakukohde/${hakukohdeOid}`,
    response: merge(hakukohde(), testHakukohdeFields),
  });

  edit
    ? cy
        .route({
          method: 'POST',
          url: '**/hakukohde',
          response: {
            muokattu: false,
          },
        })
        .as('updateHakukohdeRequest')
    : cy
        .route({
          method: 'PUT',
          url: '**/hakukohde',
          response: {
            oid: hakukohdeOid,
          },
        })
        .as('createHakukohdeRequest');

  cy.visit(
    edit
      ? `/organisaatio/${organisaatioOid}/hakukohde/${hakukohdeOid}/muokkaus`
      : `/organisaatio/${organisaatioOid}/toteutus/${toteutusOid}/haku/${hakuOid}/hakukohde`
  );
};

export const stubHakukohdeFormRoutes = ({ organisaatioOid, hakuOid }) => {
  stubCommonRoutes();

  stubKoodistoRoute({ koodisto: 'pohjakoulutusvaatimustoinenaste' });
  stubKoodistoRoute({ koodisto: 'kausi' });
  stubKoodistoRoute({ koodisto: 'valintakokeentyyppi' });
  stubKoodistoRoute({ koodisto: 'liitetyypitamm' });
  stubKoodistoRoute({ koodisto: 'kausi' });
  stubKoodiRoute(createKoodi({ koodisto: 'posti', versio: 2 }));
  stubKoodiRoute(createKoodi({ koodisto: 'posti', versio: 2 }));
  stubKoodiRoute(
    createKoodi({ koodisto: 'posti', koodiArvo: '00350', versio: 1 })
  );

  cy.route({
    method: 'GET',
    url: `**/haku/${hakuOid}`,
    response: merge(haku(), {
      oid: hakuOid,
      organisaatioOid: organisaatioOid,
      hakulomaketyyppi: 'muu',
      tila: 'julkaistu',
    }),
  });

  cy.route({
    method: 'GET',
    url: `**/hakukohde/list**`,
    response: [],
  });

  stubHakemuspalveluLomakkeetRoute();
  stubOppijanumerorekisteriHenkiloRoute();
};
