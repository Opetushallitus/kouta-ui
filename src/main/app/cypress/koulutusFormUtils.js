import merge from 'lodash/merge';

import { stubKoodistoRoute } from './utils';
import organisaatioHierarkia from './data/organisaatioHierarkia';
import organisaatio from './data/organisaatio';
import koodisto from './data/koodisto';

export const stubKoulutusFormRoutes = ({ cy, organisaatioOid }) => {
  cy.server();

  cy.route({
    method: 'GET',
    url: `**/organisaatio-service/rest/organisaatio/v4/hierarkia/hae?oid=${organisaatioOid}**`,
    response: organisaatioHierarkia({ rootOid: organisaatioOid }),
  });

  cy.route({
    method: 'GET',
    url: `**/organisaatio-service/rest/organisaatio/v4/${organisaatioOid}**`,
    response: merge(organisaatio(), {
      oid: organisaatioOid,
    }),
  });

  cy.route({
    method: 'POST',
    url: '**/organisaatio-service/rest/organisaatio/v4/findbyoids',
    response: [
      merge(organisaatio(), {
        oid: organisaatioOid,
      }),
    ],
  });

  cy.route({
    method: 'GET',
    url:
      '**/koodisto-service/rest/json/relaatio/sisaltyy-ylakoodit/koulutustyyppi_*',
    response: koodisto({ koodisto: 'koulutus' }),
  });

  cy.route({
    method: 'GET',
    url: `**/koulutus/list?organisaatioOid=${organisaatioOid}`,
    response: [],
  });

  cy.route({
    method: 'GET',
    url: '**/koodisto-service/rest/codeelement/koulutus_0',
    response: [
      {
        koodiUri: 'koulutus_0',
        versio: 1,
        metadata: [{ kieli: 'fi', nimi: 'Nimi' }],
      },
    ],
  });

  cy.route({
    method: 'GET',
    url:
      '**/eperusteet-service/api/perusteet?tuleva=true&siirtyma=false&voimassaolo=true&poistunut=false&kieli=fi&koulutuskoodi=koulutus_0',
    response: {
      data: [
        {
          kuvaus: { fi: 'koulutus_0 kuvaus' },
          osaamisalat: [],
          tutkintonimikeKoodiUri: 'nimike_1#1',
          id: '1',
        },
      ],
    },
  });

  cy.route({
    method: 'GET',
    url:
      '**/koodisto-service/rest/json/relaatio/sisaltyy-alakoodit/koulutus_0**',
    response: [
      {
        koodiUri: 'okmohjauksenala_1',
        metadata: [{ kieli: 'fi', nimi: 'Ala' }],
      },
      {
        koodiUri: 'opintojenlaajuus_1',
        metadata: [{ kieli: 'fi', nimi: '180' }],
      },
      {
        koodiUri: 'opintojenlaajuusyksikko_1',
        metadata: [{ kieli: 'fi', nimi: 'op' }],
      },
    ],
  });

  stubKoodistoRoute({ koodisto: 'tutkintonimikekk', cy });
  stubKoodistoRoute({ koodisto: 'opintojenlaajuus', cy });
  stubKoodistoRoute({ koodisto: 'koulutuksenjarjestamisenlisaosiot', cy });
};
