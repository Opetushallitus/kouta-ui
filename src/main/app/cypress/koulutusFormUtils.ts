import { merge } from 'lodash';

import {
  stubKoodistoRoute,
  stubOppijanumerorekisteriHenkiloRoute,
  stubEPerusteetByKoulutuskoodiRoute,
  stubCommonRoutes,
} from '#/cypress/utils';
import organisaatioHierarkia from '#/cypress/data/organisaatioHierarkia';
import organisaatio from '#/cypress/data/organisaatio';
import koodisto from '#/cypress/data/koodisto';
import koodistoKoulutus351107 from './data/koodistoKoulutus351107';
import koodistoKoulutukset from './data/koodistoKoulutukset';
import koodistoKoulutus351107Alakoodit from './data/koodistoKoulutus351107Alakoodit';
import koodistoKoulutus309902 from './data/koodistoKoulutus309902';
import koodistoKoulutus309902Alakoodit from './data/koodistoKoulutus309902Alakoodit';
import koodistoKoulutus671112 from './data/koodistoKoulutus671112';
import koodistoKoulutus671112Alakoodit from './data/koodistoKoulutus671112Alakoodit';
import ePerustePerusteenosat6777674 from './data/ePerustePerusteenosat6777674';
import ePerustePerusteenosat6778360 from './data/ePerustePerusteenosat6778360';
import ePeruste6777660TutkinnonOsat from './data/ePeruste6777660TutkinnonOsat';
import ePeruste6777660Rakenne from './data/ePeruste6777660Rakenne';
import ePeruste6777660 from './data/ePeruste6777660';

export const stubKoulutusFormRoutes = ({ organisaatioOid }) => {
  stubCommonRoutes();

  cy.route({
    method: 'GET',
    url: `**/organisaatio-service/rest/organisaatio/v4/hierarkia/hae**`,
    response: organisaatioHierarkia({ rootOid: organisaatioOid }),
  });

  cy.route({
    method: 'GET',
    url: `**/organisaatio-service/rest/organisaatio/v4/${organisaatioOid}**`,
    response: merge(organisaatio(), {
      oid: organisaatioOid,
    }),
  });

  cy.route({
    method: 'POST',
    url: '**/organisaatio-service/rest/organisaatio/v4/findbyoids',
    response: [
      merge(organisaatio(), {
        oid: organisaatioOid,
      }),
    ],
  });

  cy.route({
    method: 'GET',
    url:
      '**/koodisto-service/rest/json/relaatio/sisaltyy-ylakoodit/koulutustyyppi_*',
    response: koodisto({ koodisto: 'koulutus' }),
  });

  cy.route({
    method: 'GET',
    url: `**/koulutus/list?organisaatioOid=${organisaatioOid}`,
    response: [],
  });

  cy.route({
    method: 'GET',
    url: `**/koulutus/*/toteutukset/list**`,
    response: [],
  });

  cy.route({
    method: 'GET',
    url: '**/koodisto-service/rest/codeelement/koulutus_351107/**',
    response: koodistoKoulutus351107,
  });

  cy.route({
    method: 'GET',
    url:
      '**/koodisto-service/rest/json/relaatio/sisaltyy-alakoodit/koulutus_351107**',
    response: koodistoKoulutus351107Alakoodit,
  });

  cy.route({
    method: 'GET',
    url: '**/koodisto-service/rest/codeelement/koulutus_309902/**',
    response: koodistoKoulutus309902,
  });

  cy.route({
    method: 'GET',
    url:
      '**/koodisto-service/rest/json/relaatio/sisaltyy-alakoodit/koulutus_309902**',
    response: koodistoKoulutus309902Alakoodit,
  });

  cy.route({
    method: 'GET',
    url: '**/koodisto-service/rest/codeelement/koulutus_671112/**',
    response: koodistoKoulutus671112,
  });

  cy.route({
    method: 'GET',
    url:
      '**/koodisto-service/rest/json/relaatio/sisaltyy-alakoodit/koulutus_671112**',
    response: koodistoKoulutus671112Alakoodit,
  });

  cy.route({
    method: 'GET',
    url: '**/koodisto-service/rest/json/koulutus/koodi?onlyValidKoodis=true**',
    response: koodistoKoulutukset,
  });

  cy.route({
    method: 'GET',
    url: '**/eperusteet-service/api/perusteenosat/6777674',
    response: ePerustePerusteenosat6777674,
  });

  cy.route({
    method: 'GET',
    url: '**/eperusteet-service/api/perusteenosat/6778360',
    response: ePerustePerusteenosat6778360,
  });

  cy.route({
    method: 'GET',
    url: '**/eperusteet-service/api/perusteenosat/6778360',
    response: ePerustePerusteenosat6778360,
  });

  cy.route({
    method: 'GET',
    url: '**/tutkinnonosat',
    response: ePeruste6777660TutkinnonOsat,
  });

  cy.route({
    method: 'GET',
    url:
      '**/eperusteet-service/api/perusteet/6777660/suoritustavat/reformi/rakenne',
    response: ePeruste6777660Rakenne,
  });

  cy.route({
    method: 'GET',
    url: '**/eperusteet-service/api/perusteet/6777660',
    response: ePeruste6777660,
  });

  stubEPerusteetByKoulutuskoodiRoute();

  stubKoodistoRoute({ koodisto: 'tutkintonimikekk' });
  stubKoodistoRoute({ koodisto: 'opintojenlaajuus' });
  stubKoodistoRoute({ koodisto: 'koulutuksenlisatiedot' });

  stubKoodistoRoute({
    koodisto: 'kansallinenkoulutusluokitus2016koulutusalataso2',
  });

  stubOppijanumerorekisteriHenkiloRoute();
};
