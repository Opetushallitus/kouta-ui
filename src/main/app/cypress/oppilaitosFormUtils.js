import merge from 'lodash/merge';

import createKoodi from './data/koodi';
import createOrganisaatio from './data/organisaatio';
import createOrganisaatioHierarkia from './data/organisaatioHierarkia';
import { stubKoodiRoute, stubKoodistoRoute } from './utils';

export const stubOppilaitosFormRoutes = ({ organisaatioOid }) => {
  stubKoodiRoute(createKoodi({ koodisto: 'posti', versio: 2 }));

  stubKoodistoRoute({ koodisto: 'koulutuksenlisatiedot', cy });

  cy.route({
    method: 'POST',
    url: '**/organisaatio-service/rest/organisaatio/v4/findbyoids',
    response: [
      merge(createOrganisaatio(), {
        oid: organisaatioOid,
      }),
    ],
  });

  cy.route({
    method: 'GET',
    url: `**/organisaatio-service/rest/organisaatio/v4/hierarkia/hae?oid=${organisaatioOid}**`,
    response: createOrganisaatioHierarkia({ rootOid: organisaatioOid }),
  });
};
