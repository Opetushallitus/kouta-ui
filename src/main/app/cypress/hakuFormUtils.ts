import { playMocks } from 'kto-ui-common/cypress/mockUtils';
import { merge } from 'lodash/fp';

import organisaatio from '#/cypress/data/organisaatio';
import organisaatioHierarkia from '#/cypress/data/organisaatioHierarkia';
import hakuMocks from '#/cypress/mocks/haku.mocks.json';

import {
  stubHakemuspalveluLomakkeetRoute,
  stubOppijanumerorekisteriHenkiloRoute,
  stubCommonRoutes,
} from './utils';

export const stubHakuFormRoutes = ({ organisaatioOid }) => {
  stubCommonRoutes();
  playMocks(hakuMocks);

  cy.intercept(
    {
      method: 'GET',
      url: `**/organisaatio-service/rest/organisaatio/v4/hierarkia/hae**oid=${organisaatioOid}**`,
    },
    { body: organisaatioHierarkia({ rootOid: organisaatioOid }) }
  );

  cy.intercept(
    {
      method: 'GET',
      url: `**/organisaatio-service/rest/organisaatio/v4/hierarkia/hae**`,
    },
    { body: organisaatioHierarkia({ rootOid: organisaatioOid }) }
  );

  cy.intercept(
    {
      method: 'GET',
      url: `**/organisaatio-service/rest/organisaatio/v4/${organisaatioOid}**`,
    },
    {
      body: merge(organisaatio(), {
        oid: organisaatioOid,
      }),
    }
  );

  cy.intercept(
    {
      method: 'POST',
      url: '**/organisaatio-service/rest/organisaatio/v4/findbyoids',
    },
    {
      body: [
        merge(organisaatio(), {
          oid: organisaatioOid,
        }),
      ],
    }
  );

  cy.intercept({ method: 'GET', url: '**/haku/list**' }, { body: [] });

  cy.intercept({ method: 'GET', url: `**/toteutus/list**` }, { body: [] });

  cy.intercept({ method: 'GET', url: `**/koulutus/list**` }, { body: [] });

  cy.intercept({ method: 'GET', url: `**/search/haku/**` }, { body: [] });

  stubHakemuspalveluLomakkeetRoute();
  stubOppijanumerorekisteriHenkiloRoute();
};
