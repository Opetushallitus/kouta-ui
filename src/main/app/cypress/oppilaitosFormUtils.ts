import { playMocks } from 'kto-ui-common/cypress/mockUtils';
import { merge } from 'lodash/fp';

import createOrganisaatio from '#/cypress/data/organisaatio';
import createOrganisaatioHierarkia from '#/cypress/data/organisaatioHierarkia';
import oppilaitosMocks from '#/cypress/mocks/oppilaitos.mocks.json';
import { stubCommonRoutes } from '#/cypress/utils';

export const stubOppilaitosFormRoutes = ({ organisaatioOid }) => {
  stubCommonRoutes();
  playMocks(oppilaitosMocks);

  cy.intercept(
    {
      method: 'POST',
      url: '**/organisaatio-service/rest/organisaatio/v4/findbyoids',
    },
    {
      body: [
        merge(createOrganisaatio(), {
          oid: organisaatioOid,
        }),
      ],
    }
  );

  cy.intercept(
    {
      method: 'GET',
      url: `**/organisaatio-service/rest/organisaatio/v4/hierarkia/hae**oid=${organisaatioOid}**`,
    },
    { body: createOrganisaatioHierarkia({ rootOid: organisaatioOid }) }
  );
};
