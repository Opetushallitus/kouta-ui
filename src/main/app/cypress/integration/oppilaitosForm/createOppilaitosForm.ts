import { stubOppilaitosFormRoutes } from '#/cypress/oppilaitosFormUtils';
import {
  fillAsyncSelect,
  selectOption,
  typeToEditor,
  getByTestId,
  jatka,
  paste,
  fillKieliversiotSection,
  fillTilaSection,
  tallenna,
} from '#/cypress/utils';

const fillPerustiedotSection = () => {
  getByTestId('perustiedotSection').within(() => {
    getByTestId('opiskelijoita').find('input').pipe(paste('1'));
    getByTestId('korkeakouluja').find('input').pipe(paste('2'));
    getByTestId('tiedekuntia').find('input').pipe(paste('3'));
    getByTestId('kampuksia').find('input').pipe(paste('4'));
    getByTestId('yksikoita').find('input').pipe(paste('5'));
    getByTestId('toimipisteita').find('input').pipe(paste('6'));
    getByTestId('akatemioita').find('input').pipe(paste('7'));

    jatka();
  });
};

const fillEsittelySection = () => {
  getByTestId('esittelySection').within(() => {
    typeToEditor('Esittely');

    jatka();
  });
};

const fillTeemakuvaSection = () => {
  getByTestId('teemakuvaSection').within(() => {
    jatka();
  });
};

const fillTietoaOpiskelustaSection = () => {
  getByTestId('tietoaSection').within(() => {
    selectOption('Opintojen rahoitus');
    typeToEditor('Tietoa');
    jatka();
  });
};

const fillYhteystiedotSection = () => {
  getByTestId('yhteystiedotSection').within(() => {
    getByTestId('lisaaYhteystietoButton').click({ force: true });
    cy.findByLabelText(/oppilaitoslomake\.yhteystiedonNimi/).pipe(
      paste('Yhteystiedon nimi')
    );

    cy.findByLabelText(/yleiset.postiosoite/).pipe(paste('Osoite'));

    // NOTE: postinumeros are with testIds for they share the same translationkey
    getByTestId('postinumero').within(() => {
      fillAsyncSelect('00350');
    });

    cy.findByLabelText(/yleiset.kayntiosoite/).pipe(paste('Osoite'));

    getByTestId('kayntiosoitePostinumero').within(() => {
      fillAsyncSelect('00350');
    });

    cy.findByLabelText(/yleiset.sahkoposti/).pipe(
      paste('sahkoposti@sahkoposti.fi')
    );

    cy.findByLabelText(/yleiset.puhelinnumero/).pipe(paste('12345'));

    cy.findByLabelText(/yleiset.verkkosivu/).pipe(paste('www.verkkosivu.fi'));

    jatka();
  });
};

const skipHakijapalveluidenYhteystiedotSection = () => {
  getByTestId('hakijapalveluidenYhteystiedotSection').within(() => {
    jatka();
  });
};

const fillHakijapalveluidenYhteystiedotSection = () => {
  getByTestId('hakijapalveluidenYhteystiedotSection').within(() => {
    cy.findByLabelText(/oppilaitoslomake\.yhteystiedonNimi/).pipe(
      paste('Testihakijapalvelu')
    );

    cy.findByLabelText(/yleiset.postiosoite/).pipe(paste('Osoite'));

    // NOTE: postinumeros are with testIds for they share the same translationkey
    getByTestId('postinumero').within(() => {
      fillAsyncSelect('00350');
    });

    cy.findByLabelText(/yleiset.kayntiosoite/).pipe(paste('Osoite'));

    getByTestId('kayntiosoitePostinumero').within(() => {
      fillAsyncSelect('00350');
    });

    cy.findByLabelText(/yleiset.sahkoposti/).pipe(
      paste('sahkoposti@sahkoposti.fi')
    );

    cy.findByLabelText(/yleiset.puhelinnumero/).pipe(paste('12345'));

    cy.findByLabelText(/yleiset.verkkosivu/).pipe(paste('www.verkkosivu.fi'));

    jatka();
  });
};

const passOsatSection = () => {
  getByTestId('oppilaitoksenOsatSection').within(() => {
    jatka();
  });
};

export const createOppilaitosForm = () => {
  const organisaatioOid = '1.1.1.1.1.1';

  beforeEach(() => {
    stubOppilaitosFormRoutes({ organisaatioOid });

    cy.intercept(
      { method: 'GET', url: `**/oppilaitos/${organisaatioOid}` },
      { body: '' }
    );

    cy.visit(`/organisaatio/${organisaatioOid}/oppilaitos`);
  });

  it('should be able to create oppilaitos without hakijapalveluyhteystiedot', () => {
    cy.intercept(
      { method: 'PUT', url: '**/oppilaitos' },
      {
        body: {
          oid: '1.2.3.4.5.6',
        },
      }
    ).as('createOppilaitosResponse');

    fillKieliversiotSection({ jatka: true });

    fillPerustiedotSection();

    fillEsittelySection();

    fillTeemakuvaSection();

    passOsatSection();

    fillTietoaOpiskelustaSection();

    fillTilaSection();

    fillYhteystiedotSection();

    skipHakijapalveluidenYhteystiedotSection();

    tallenna();

    cy.wait('@createOppilaitosResponse').then(({ request }) => {
      cy.wrap(request.body).toMatchSnapshot();
    });
  });

  it('should be able to create oppilaitos with hakijapalveluyhteystiedot', () => {
    cy.intercept(
      { method: 'PUT', url: '**/oppilaitos' },
      {
        body: {
          oid: '1.2.3.4.5.6',
        },
      }
    ).as('createOppilaitosResponse');

    fillKieliversiotSection({ jatka: true });

    getByTestId('yhteystiedotSection').click();

    fillYhteystiedotSection();

    fillHakijapalveluidenYhteystiedotSection();

    tallenna();

    cy.wait('@createOppilaitosResponse').then(({ request }) => {
      cy.wrap(request.body).toMatchSnapshot();
    });
  });
};
