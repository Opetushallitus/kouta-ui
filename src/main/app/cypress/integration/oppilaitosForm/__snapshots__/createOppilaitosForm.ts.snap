exports[`createOppilaitosForm > should be able to create oppilaitos #0`] =
{
  "kielivalinta": [
    "fi"
  ],
  "logo": null,
  "metadata": {
    "akatemioita": 7,
    "esittely": {
      "fi": "<p>Esittely</p>"
    },
    "kampuksia": 4,
    "korkeakouluja": 2,
    "opiskelijoita": 1,
    "tiedekuntia": 3,
    "tietoaOpiskelusta": [
      {
        "otsikkoKoodiUri": "organisaationkuvaustiedot_0#1",
        "teksti": {
          "fi": "Tietoa"
        }
      }
    ],
    "toimipisteita": 6,
    "yhteystiedot": {
      "osoite": {
        "osoite": {
          "fi": "Osoite"
        },
        "postinumeroKoodiUri": "posti_0#2"
      },
      "puhelinnumero": {
        "fi": "12345"
      },
      "sahkoposti": {
        "fi": "sahkoposti@sahkoposti.fi"
      },
      "wwwSivu": {
        "fi": "www.verkkosivu.fi"
      }
    },
    "yksikoita": 5
  },
  "muokkaaja": "1.2.246.562.24.62301161440",
  "oid": "1.1.1.1.1.1",
  "organisaatioOid": "1.1.1.1.1.1",
  "teemakuva": null,
  "tila": "julkaistu"
};
